src/main/java/com/gmail/inverseconduit/commands/sets/AdminCommands.java

```diff
@@ -27,14 +16,7 @@ public static CommandHandle addAdminCommand(BotConfig bc) {
 			String[] args = message.getMessage().trim().split(" ");
 			try {
 				Long newUserId = Long.parseLong(args[1]);
-				if (Permissions.isAdmin(newUserId, bc))
-					Permissions.addAdmin(newUserId, bc);
-				else {
-					return "User already has elevated privileges!";
-				}
-				Path file = Paths.get("bot.properties");
-				addAdminToFile(file, (long) newUserId);
-				return "Added userID " + newUserId + " successful!";
+				return Permissions.addAdmin(newUserId, bc);

```

Comment:
no. adding an admin shouldn't return a String for the user, just a `Result` or something similar. The result can be examined to create a String that's intended for the user:

    switch (Permissions.addAdmin(newUserId, bc)) {
         case ALREADY_EXISTS:
              return "User already has elevated privileges";
         case SUCCESS:
              return "Added user " + newUserId + " successfully!";
         case FAILURE:
              return "Failed to add admin. Please contact the bot owner";
       }